### PR TITLE
add KMT test washer

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Add KMT support for Test Washer (added in https://github.com/DataDog/datadog-agent/pull/25015)

### Motivation

Better handling of flaky tests

### Additional Notes

https://datadoghq.atlassian.net/browse/EBPF-523

### CODE REVIEW

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components

**Review by pducolin — APPROVED** [review]
**👏 praise**
Nice, thank you!

File: pkg/util/testutil/flake/parse_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsFlaky(t *testing.T) {
+	kf := &KnownFlakyTests{}
+	kf.Add("", "TestEKSSuite/TestCPU")
+
+	assert.True(t, kf.IsFlaky("", "TestEKSSuite/TestCPU/TestCPUUtilization"))
+	assert.True(t, kf.IsFlaky("", "TestEKSSuite"))
+	assert.False(t, kf.IsFlaky("", "TestEKSSuite/TestMemory"))
```
Comment (pducolin) [inline pkg/util/testutil/flake/parse_test.go:20]: **💬 suggestion**
Adding a couple of test examples to increase coverage
```suggestion
	assert.False(t, kf.IsFlaky("", "TestEKSSuite/TestMemory"))
	assert.True(t, kf.IsFlaky("", "TestEKSSuite/TestCPU"))
	assert.False(t, kf.IsFlaky("", "TestECSSuite/TestCPU"))
```

**Review by amitslavin — APPROVED** [review]

File: pkg/util/testutil/flake/parse.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	if err := dec.Decode(&pkgToTests); err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+	kf := &KnownFlakyTests{packageTestList: make(map[string]map[string]struct{})}
+	for pkg, tests := range pkgToTests {
+		testSet := make(map[string]struct{})
+		for _, t := range tests {
+			testSet[t] = struct{}{}
+		}
+		kf.packageTestList[pkg] = testSet
```
Comment (pducolin) [inline pkg/util/testutil/flake/parse.go:75]: **❓ question**
Why not using 
```suggestion
		kf.Add(pkg, testSet)
```
instead?

**Review by gjulianm — APPROVED** [review]

File: pkg/util/testutil/flake/parse.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	for test := range k.packageTestList[pkg] {
+		if strings.HasPrefix(test, testName+"/") {
+			return true
+		}
+	}
+	return false
+}
+
+// Parse parses the reader in the flake.yaml format
+func Parse(r io.Reader) (*KnownFlakyTests, error) {
```
Comment (pducolin) [inline pkg/util/testutil/flake/parse.go:63]: **❓ question**
Can you add an extra test for this function? There should be one in our python task

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 4]
```diff
@@ -1223,7 +1224,7 @@ func testPostgresProtocolClassificationWrapper(enableTLS bool) func(t *testing.T
 func testPostgresProtocolClassification(t *testing.T, tr *Tracer, clientHost, targetHost, serverHost string, enableTLS bool) {
 	skippers := []func(t *testing.T, ctx testContext){skipIfUsingNAT}
 	if enableTLS {
-		t.Skip("TLS+Postgres classification tests are flaky")
+		flake.Mark(t)
```
Comment (guyarb) [inline pkg/network/tracer/tracer_usm_linux_test.go:1227]: what does it mean?
when will the test run?

File: test/new-e2e/system-probe/test-json-review/main_test.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+{"Time":"2024-06-14T22:24:53.156263319Z","Action":"output","Package":"a/b/c","Test":"testname","Output":"=== RUN   testname\n"}
+{"Time":"2024-06-14T22:24:53.156271614Z","Action":"output","Package":"a/b/c","Test":"testname","Output":"    file_test.go:10: flakytest: this is a known flaky test\n"}
+{"Time":"2024-06-14T22:26:02.039003529Z","Action":"fail","Package":"a/b/c","Test":"testname","Elapsed":26.25}
+`
+
+func TestFlakeInOutput(t *testing.T) {
+	out, err := reviewTestsReaders(bytes.NewBuffer([]byte(flakeTestData)), nil)
+	require.NoError(t, err)
+	assert.Empty(t, out.Failed)
+	assert.NotEmpty(t, out.Flaky)
```
Comment (GustavoCaso) [inline test/new-e2e/system-probe/test-json-review/main_test.go:28]: Should we make the assertion check the contents of `out.Flaky`? 

Also, should we add a test case for failed and reruns?

File: pkg/network/tracer/tracer_usm_linux_test.go (hunk 4 above)
Comment (guyarb) [reply pkg/network/tracer/tracer_usm_linux_test.go:1227]: @pducolin I guess you'll know best
I do see that on main we will still run the tests (even though the tests are flaky)
does it mean those tests will be reported as flaky in the datadog CI tests metrics?

Comment (brycekahle) [reply pkg/network/tracer/tracer_usm_linux_test.go:1227]: The tests still run, but they will not cause the job to fail. The same data is still reported to CI Visibility, so you can see failures/passes and whether or not a test is flaky.

File: pkg/util/testutil/flake/parse.go (hunk 2 above)
Comment (brycekahle) [reply pkg/util/testutil/flake/parse.go]: Changed

File: pkg/util/testutil/flake/parse.go (hunk 3 above)
Comment (brycekahle) [reply pkg/util/testutil/flake/parse.go:63]: Added

File: test/new-e2e/system-probe/test-json-review/main_test.go (hunk 5 above)
Comment (brycekahle) [reply test/new-e2e/system-probe/test-json-review/main_test.go]: Added

File: pkg/network/tracer/tracer_usm_linux_test.go (hunk 4 above)
Comment (guyarb) [reply pkg/network/tracer/tracer_usm_linux_test.go:1227]: Then please exclude usm from the change, I don't want those tests to be reported as flaky as they will cause noise for us

**Pull request merged**